### DISCUSSION TITLE
Signing url limit/efficiency

### DISCUSSION THREAD

**Discussion by cjacky475** [supabase/supabase#22172] [Questions]
Hello, 

in my app users can have items. Items are visible to the owner and friends. Each item can have multiple images. All the images are in a private bucket stored within a folder of an item ID.

Are there any limitation for me to generate a lot of presigned URLs (for storage) when user scrolls through friend's items? As I understand I would need to generate a presigned URL for each picture and send them within a request from backend to frontend. Is this optimal solution for my problem?   

Thank you.

Jacky

**Comment by GaryAustin1** [comment]
If you require a private bucket (versus hard to guess) then signed URL's are your option.  There is a call to generate several at once. https://supabase.com/docs/reference/javascript/storage-from-createsignedurls
